src/raw/document_buf.rs

```diff
@@ -188,7 +188,6 @@ impl RawDocumentBuf {
     /// result in errors when communicating with MongoDB.
     ///
     /// If the provided key contains an interior null byte, this method will panic.
```

Comment:
Can we add some language here about the acceptable input types and show a borrowed value being appended in the code example?